Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -119,9 +162,56 @@ private Vector3 WorldSpacePressDirection
                 return transform.forward;
             }
         }
+        private Transform PushSpaceSourceTransform
+        {
+            get { return movingButtonVisuals != null ? movingButtonVisuals.transform : transform; }
+        }
+
+        private float LocalToWorldScale
+        {
+            get
+            {
+                return transform.TransformVector(WorldSpacePressDirection).magnitude;
+            }
+        }
+        private Vector3 LocalForward
+        {
+            get
+            {
+                var nearInteractionTouchable = GetComponent<NearInteractionTouchable>();
+                return (nearInteractionTouchable != null) ? -1.0f * nearInteractionTouchable.LocalForward : Vector3.forward;
+            }
+        }
+
+        private Vector3 initialPosition;
+        /// <summary>
+        /// The position from where the button starts to move. 
+        /// </summary>
+        private Vector3 InitialPosition
+        {
+            get
+            {
+                if (Application.isPlaying) // we're using a cached position in play mode as the moving visuals will be moved during button interaction
+                {
+                    return initialPosition;
+                }
+                else
+                {
+                    return PushSpaceSourceTransform.position;
+                }
+
+            }
+        }
 
         #endregion
 
+
+        private void Awake()
+        {
+            currentPushDistance = startPushDistance;
```

Comment:
Do we handle the case where a button gets disabled, moved, and then enabled? Does currentPushDistance get properly re-initialized / zeroed?

Reply:
good point - this is not handled right now

Reply:
we can actually move the content of this function into OnEnable() that should solve the issue - we don't need to do this in Awake()